vortex-array/src/view.rs

```diff
@@ -126,7 +126,8 @@ impl ArrayView {
             .iter()
             .take(idx)
             .map(|child| Self::cumulative_nbuffers(child))
-            .sum();
+            .sum::<usize>()
+            + self.has_buffer() as usize;
```

Comment:
fun!

Reply:
The bug here is that an array has buffers of the children plus optionally it's own buffer. So we were taking 0th buffer when we were asking for 0th child without accounting for the own buffer which would be the first buffer

Reply:
You can see this 1 off offset in cumulative_nbuffers